ibm/service/power/data_source_ibm_pi_network_interfaces.go

```diff
@@ -0,0 +1,166 @@
+			},
+			// Attributes
+			Attr_Interfaces: {
+				Computed:    true,
+				Description: "Network Interfaces.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_CRN: {
+							Computed:    true,
+							Description: "The Network Interface's crn.",
+							Type:        schema.TypeString,
+						},
+						Attr_ID: {
+							Computed:    true,
+							Description: "The unique Network Interface ID.",
+							Type:        schema.TypeString,
+						},
+						Attr_Instance: {
+							Computed:    true,
+							Description: "The attached instance to this Network Interface.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									Attr_Href: {
+										Computed:    true,
+										Description: "Link to instance resource.",
+										Type:        schema.TypeString,
+									},
+									Attr_InstanceID: {
+										Computed:    true,
+										Description: "The attached instance ID.",
+										Type:        schema.TypeString,
+									},
+								},
+							},
+							Type: schema.TypeList,
+						},
+						Attr_IPAddress: {
+							Computed:    true,
+							Description: "The ip address of this Network Interface.",
+							Type:        schema.TypeString,
+						},
+						Attr_MacAddress: {
+							Computed:    true,
+							Description: "The mac address of the Network Interface.",
+							Type:        schema.TypeString,
+						},
+						Attr_Name: {
+							Computed:    true,
+							Description: "Name of the Network Interface (not unique or indexable).",
+							Type:        schema.TypeString,
+						},
+						Attr_NetworkSecurityGroupID: {
+							Computed:    true,
+							Description: "ID of the Network Security Group the network interface will be added to.",
+							Type:        schema.TypeString,
+						},
+						Attr_Status: {
+							Computed:    true,
+							Description: "The status of the network address group.",
+							Type:        schema.TypeString,
+						},
+						Attr_UserTags: {
+							Computed:    true,
+							Description: "List of user tags attached to the resource.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Set:         schema.HashString,
+							Type:        schema.TypeSet,
+						},
+					},
+				},
+				Type: schema.TypeList,
+			},
+		},
+	}
+}
+
+func dataSourceIBMPINetworkInterfacesRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	networkID := d.Get(Arg_NetworkID).(string)
+	networkC := instance.NewIBMPINetworkClient(ctx, sess, cloudInstanceID)
+	networkInterfaces, err := networkC.GetAllNetworkInterfaces(networkID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var genID, _ = uuid.GenerateUUID()
+	d.SetId(genID)
+	interfaces := []map[string]interface{}{}
+	if networkInterfaces.Interfaces != nil {
```

Comment:
Always use len() which checks for nil + empty

Reply:
Updated